tools/main.py

```diff
@@ -0,0 +1,131 @@
+import msgpack
+import csv
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from Packet import *
+
+
+def retrieve_data(infile, boards, start=-1, end=float('inf')):
+    # Takes logfile and a list of boards' data to read
+    # Returns dictionary, 'data', with information for all boards using timestamp as keys
+    # Will only return data within specific timestamp if specified
+
+    data = dict()
+    previous_timestamp = 0
+    data[previous_timestamp] = {}
+
+    # Creates first item in data, at time=0 and values=None for each board
+    for board in boards:
+        data[previous_timestamp][board] = None
+
+    # Loop through log file to retrieve data, adds it to 'data'
+    # Turns each logged packet into a Packet object for simplicity.
+    infile.seek(0)
+    for raw_packet in msgpack.Unpacker(infile):
+        for packet in Packet.sort(raw_packet):
+
+            # Check if packet is within specified timerange
+            if packet.timestamp <= start or packet.timestamp >= end:
+                continue
+
+            # Check if packet is of a specified board
+            elif packet.name in boards:
+                # initialize the key
+                if packet.timestamp not in data:
+                    data[packet.timestamp] = {}
+
+                data[packet.timestamp][packet.name] = packet.value
+
+                # To prevent 'holes' in the dictionary, all other values are
+                # copied over from the previous timestamp
+                for board in boards:
+                    if board not in data[packet.timestamp]:
+                        data[packet.timestamp][board] = data[previous_timestamp][board]
+
+                previous_timestamp = packet.timestamp
+
+    # Remove the initial entry
+    del data[0]
+
+    data_list = list()
+    for timestamp, items in data.items():
+        data[timestamp]['Timestamp'] = timestamp
+        data_list.append(items)
+    del data
+
+    return data_list
+
+
+def to_csv(outfile, field_names, data):
```

Comment:
use pandas for this instead of your own helper function

Reply:
not ideal in this case